extensions/control-plane/data-plane-transfer/src/main/java/org/eclipse/edc/connector/dataplane/transfer/flow/DataPlaneTransferDataFlowController.java

```diff
@@ -24,44 +24,41 @@
 import org.eclipse.edc.spi.response.StatusResult;
 import org.eclipse.edc.spi.types.domain.DataAddress;
 import org.eclipse.edc.spi.types.domain.transfer.DataFlowRequest;
-import org.eclipse.edc.util.string.StringUtils;
 import org.jetbrains.annotations.NotNull;
 
 import java.util.UUID;
 
 import static org.eclipse.edc.connector.dataplane.transfer.spi.DataPlaneTransferConstants.HTTP_PROXY;
 
-/**
- * Implementation of {@link DataFlowController} that delegates data transfer to Data Plane instance.
- * Note that Data Plane can be embedded in current runtime (test, samples...) or accessed remotely.
- * The present {@link DataFlowController} is triggered when destination type in the {@link DataRequest} is different from
- * {@link DataPlaneTransferConstants#HTTP_PROXY}, as this one is reserved for synchronous data transfers.
- */
-public class DataPlaneTransferFlowController implements DataFlowController {
-    private final DataPlaneTransferClient client;
+public class DataPlaneTransferDataFlowController implements DataFlowController {
+
+    private final DataProxyService dataPlaneProxyService;
     private final ControlPlaneApiUrl callbackUrl;
+    private final DataPlaneClient dataPlaneClient;
 
-    public DataPlaneTransferFlowController(DataPlaneTransferClient client, ControlPlaneApiUrl callbackUrl) {
-        this.client = client;
+    public DataPlaneTransferDataFlowController(DataProxyService dataPlaneProxyService, ControlPlaneApiUrl callbackUrl, DataPlaneClient dataPlaneClient) {
+        this.dataPlaneProxyService = dataPlaneProxyService;
         this.callbackUrl = callbackUrl;
+        this.dataPlaneClient = dataPlaneClient;
     }
 
     @Override
     public boolean canHandle(DataRequest dataRequest, DataAddress contentAddress) {
-        var type = dataRequest.getDestinationType();
-        if (!StringUtils.isNullOrBlank(type)) {
-            return !HTTP_PROXY.equals(dataRequest.getDestinationType());
-        }
-        return false;
+        return true;
     }
 
     @Override
     public @NotNull StatusResult<Void> initiateFlow(DataRequest dataRequest, DataAddress contentAddress, Policy policy) {
+        if (HTTP_PROXY.equals(dataRequest.getDestinationType())) {
```

Comment:
I preferred when they were two different `DataFlowController`s: one for `ProviderPush` and one for `ConsumerPull`, it showed the difference in a clearer way in my opinion.